src/map/battle.cpp

```diff
@@ -4253,7 +4264,8 @@ static int battle_calc_attack_skill_ratio(struct Damage* wd, struct block_list *
 		case KO_SETSUDAN:
 			skillratio += 100 * (skill_lv - 1);
 			RE_LVL_DMOD(100);
-			if(tsc && tsc->data[SC_SPIRIT])
+			// Bonus damage added when target is soul linked.
+			if(tsc && ( tsc->data[SC_SPIRIT] || tsc->data[SC_SOULGOLEM] || tsc->data[SC_SOULSHADOW] || tsc->data[SC_SOULFALCON] || tsc->data[SC_SOULFAIRY]))
```

Comment:
this needs to check which tsc->data[] is being used!

Reply:
Not sure if this is required. I never see any other sc->data being checked.

Reply:
i dont follow your reasoning.  if all that matters is the level of SC_SPIRIT the player gets, then what's hte point of checking the reaper statuses? they all dispel SC_SPIRIT.